src/engine/Sim3D.ts

```diff
@@ -470,75 +468,63 @@ export class Sim3D extends EventEmitter {
-      });
-      this.addWall({
-        type: "wall",
-        start: { x: worldConfig.xLength / 2, y: -worldConfig.zLength / 2 },
-        end: { x: worldConfig.xLength / 2, y: worldConfig.zLength / 2 },
-        height: worldConfig.perimeter.height,
-        thickness: worldConfig.perimeter.thickness,
-      });
-    } else if (worldConfig.walls) {
-      console.warn(
-        "Using worldConfig.walls will be deprecated in favor of the worldConfig.perimeter field"
-      );
-      if (worldConfig.walls.length === 0) {
-        this.addWall({
-          type: "wall",
-          start: { x: -worldConfig.xLength / 2, y: -worldConfig.zLength / 2 },
-          end: { x: worldConfig.xLength / 2, y: -worldConfig.zLength / 2 },
-          baseColor: 0x00ff00,
-        });
-        this.addWall({
-          type: "wall",
-          start: { x: -worldConfig.xLength / 2, y: worldConfig.zLength / 2 },
-          end: { x: worldConfig.xLength / 2, y: worldConfig.zLength / 2 },
-          baseColor: 0xff0000,
-        });
-        this.addWall({
-          type: "wall",
-          start: { x: -worldConfig.xLength / 2, y: -worldConfig.zLength / 2 },
-          end: { x: -worldConfig.xLength / 2, y: worldConfig.zLength / 2 },
-        });
-        this.addWall({
-          type: "wall",
-          start: { x: worldConfig.xLength / 2, y: -worldConfig.zLength / 2 },
-          end: { x: worldConfig.xLength / 2, y: worldConfig.zLength / 2 },
-        });
-      } else {
-        worldConfig.walls.forEach((wallSpec) => {
-          this.addWall(wallSpec);
-        });
-      }
+    const walls = wallSpecs(worldConfig);
+
+    for (const spec of walls) {
+      this.addWall(spec);
     }
 
     // Axes
     const axesHelper = new THREE.AxesHelper(1);
+    axesHelper.position.x = -worldConfig.xLength / 2 - 1;
+    axesHelper.position.z = -worldConfig.zLength / 2 - 1;
     this.scene.add(axesHelper);
+
+    // Cosmetics
+    const floorGeometry = new THREE.PlaneGeometry(100, 100, 100, 100);
+    const floorTexture = new THREE.TextureLoader().load("/assets/floor.png");
+    floorTexture.repeat = new THREE.Vector2(100, 100);
+    floorTexture.wrapS = THREE.RepeatWrapping;
+    floorTexture.wrapT = THREE.RepeatWrapping;
+    const floorMaterial = new THREE.MeshStandardMaterial({ map: floorTexture });
+    const floor = new THREE.Mesh(floorGeometry, floorMaterial);
+    floor.rotateX(-Math.PI / 2);
+    floor.position.y = -0.1;
+    this.scene.add(floor);
+
+    const blockGeometry = new THREE.BoxGeometry(1, 1, 1);
+    const redBlockMaterial = new THREE.MeshStandardMaterial({
+      color: 0xff0000,
+    });
+    const greenBlockMaterial = new THREE.MeshStandardMaterial({
+      color: 0x00ff00,
+    });
+    const blueBlockMaterial = new THREE.MeshStandardMaterial({
+      color: 0x0000ff,
+    });
+    const blockParent = new THREE.Object3D();
+    {
+      const block = new THREE.Mesh(blockGeometry, redBlockMaterial);
```

Comment:
rather than opening anonymous scopes - can we move these to helper functions? Perhaps a `createBlock` function?